channeldb/invoices.go

```diff
@@ -679,7 +692,12 @@ func fetchInvoiceNumByRef(invoiceIndex, payAddrIndex kvdb.RBucket,
 		invoiceNumByAddr []byte
 	)
 	if payAddr != nil {
-		invoiceNumByAddr = payAddrIndex.Get(payAddr[:])
+		// Only allow lookups for payment address if it is not a blank
+		// payment address, which is a special-cased value for legacy
+		// keysend invoices.
+		if *payAddr != BlankPayAddr {
```

Comment:
But do we look up legacy keysends by addr anywhere?

Reply:
i don't think so, but just in case someone has a blank pay addr in their index i thought it was good to have this check and keep the results consistent as if we had done a migration to remove it.

Reply:
Ok, got it